ui/app/src/shared-worker.ts

```diff
@@ -122,6 +124,8 @@ function retrieve() {
           log('New token received');
           status = 'active';
           current = response;
+          // Open the root topic socket.
+          openSocket({ site: null, xsrfToken });
```

Comment:
Why here? This method has to do with retrieving the JWT and it runs constantly throughout the lifecycle of the worker. Why would we kill and create the socket connection over and over?
This should be done just once when connecting to the worker or after a timeout.